snippets/javascript/jscalculator.md

```diff
@@ -0,0 +1,48 @@
+# JS based simple calculator
+
+```js
+'use strict'
+
+let operator = '';
+let operation = '';
+let number1;
+let number2;
+
+// accept input from the user and check if it is a number 
+while(isNaN(number1)) {
+ number1 = parseFloat(prompt ('Enter the first number: '));
+}
+  operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
+  while(!(operator === '+' || operator === '-' || operator === '*' || operator === '/')){
+    operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
+  }
+  // accept input from the user and check if it is a number 
+  while(isNaN(number2)) {  
+    number2 = parseFloat(prompt ('Enter the second number: '));
+   }
+
+let result; // declaration of the variable.  
+  
+// use if, elseif and else keyword to define the calculator condition in JavaScript.  
```

Comment:
Note: You could also use a switch statement for these use cases https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch 
 
 But `if-else` is fine too, you can just resolve this one without changes if you want!